docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,108 @@
+* Confusing for backporting. We will need to keep creating new labels and making sure we remove the old labels because we don't want to accidently backport to the wrong branch.
+
+### Minor release branches (current strategy)
+#### Pros:
+* Adapter CI testing will continue to work with this strategy and makes it less complicated overall. Adapters and `dbt-core` need to be on the same minor version but can be on different patch versions. The current CI workflows verify that the changes slated for a new `1.1` patch are compatible across all repos by checking out the `1.1.latest` branch of each repo. The 1:1 mapping of adapter branches to `dbt-core` branches makes this easy.
+* Easy backporting. There is no need to figure out the exact branch that you need to backport to and accidently backporting to an older, stale branch.
+#### Cons:
+* Branches contain more than 1 release which is confusing. Without a 1:1 mapping between branch and release, you must rely on tags instead to know where in the branch the release was cut.
+* Releasing from the same branch multiple times can lead to confusion as to which version we are releasing and actually stops us from releasing the wrong version.
+
+## Decisions
+
+### Version phases
+<details>
+<summary> Header Explanations: </summary>
+
+- phases: defined in [Common Terminology](#common-terminology)
+- Released?: do we plan to release this phase to GitHub, PyPi, etc.
+- Branches: the branches where these release phases are present (e.g. betas should only exist on the `main` branch)
+- Applicable Release Versions: the release versions where the release phase is applicable (i.e. we will only have alphas for major and minor versions, we will not have alphas for patch versions)
+- Expectations: the stability of the code changes in the release
+</details>
+
+| phase | Released? | Branches | Applicable Release Versions | Expectations |
+| ---- | --------- | -------- | ---------------- | ------------ |
+| Alpha | No | `main` | Major / Minor | Experimental |
+| Beta | Yes | `main` | Major / Minor | Experimental |
+| RC | Yes | Release branch | Major / Minor / Patch | Pre-production |
+| Final | Yes | Release branch | Major / Minor / Patch | Production |
+
+### Branching Strategy
+Based off of the release version and expectations, the version phase's branching strategy can be determined. 
+ * Alpha and Beta versions are experimental, and therefore exist only on the `main` branch, where all our changes reside
+ * Alpha versions are never released. They signify that development is underway, though the changes they contain have not been released in any form
+ * An RC version denotes a more stable state. A release branch will be created for RC releases to limit the changes a release contains
```

Comment:
we should probably use the term "release candidate" here even though it's implied.